### PR TITLE
Add two-factor authentication #525

### PR DESCRIPTION
**Pull request by sztamas**

### CODE REVIEW

File: rdrf/rdrf/auth/views.py [hunk 1]
```diff
@@ -6,32 +6,65 @@
...
 
 logger = logging.getLogger(__name__)
 
 
+@receiver(user_logged_in)
+def user_login_callback(sender, request=None, user=None, **kwargs):
+    if is_user_privileged(user) and default_device(user) is None:
+        link = ('<a href="%(url)s" class="alert-link">' + _('click here') + '</a>') % {'url': reverse('two_factor:setup')}
+        msg = mark_safe(
+                _('We strongly recommend that you protect your account with Two-Factor authentication. '
```
Comment (id2359) [inline rdrf/rdrf/auth/views.py]: So this message will always appear as a nag ?

Comment (sztamas) [reply rdrf/rdrf/auth/views.py]: Yes, but only for "system users".

Comment (id2359) [reply rdrf/rdrf/auth/views.py]: I found that if as an admin I made a user use 2fa and then logged in as that user I get the 2fa setup screen as expected, but after I set up the 2fa ,  am still greeted by that 2fa recommendation message ( despite having just set it up)

Comment (id2359) [reply rdrf/rdrf/auth/views.py]: Also after being "forced" to setup 2fa by an admin, the disable 2fa option still exists in the drop down, which I can then click. Then when I log back in I'm forced to set it up again.

So if 2fa is being made mandatory ( via the admin setting it as required in the admin interface), shouldn't the cancel option simply not be presented to the user at all in this case?

File: rdrf/rdrf/auth/views.py [hunk 2]
```diff
@@ -40,7 +40,7 @@
 
 @receiver(user_logged_in)
 def user_login_callback(sender, request=None, user=None, **kwargs):
-    if is_user_privileged(user) and default_device(user) is None:
+    if is_user_privileged(user) and not user.require_2_fact_auth and default_device(user) is None:
```
Comment (id2359) [inline rdrf/rdrf/auth/views.py]: looks good

**Pull request merged**